src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -27,115 +27,82 @@
-    logger.error('An exception has occurred in the process of generating the users.csv file: %s', ex)
-
-
-# Export sections
-logger.info("Exporting sections")
-sections_list = []
-try:
-    # first we need to get a list of courses
-    logger.info("Exporting sections - Getting courses")
-    courses_response = request_client.get_courses(page_size)
-    courses_list: List[Any] = []
-    while True:
-        courses_list = courses_list + courses_response.current_page_items
-        if courses_response.get_next_page() is None:
-            break
-
-    # now we can get a list of sections
-    course_ids = map(lambda x: x["id"], courses_list)
-    sections_list = request_client.get_section_by_course_ids(list(course_ids))
-    export_data.to_csv(sections_list, os.path.join(schoology_output_path, "sections.csv"))
-except Exception as ex:
-    logger.error('An exception has occurred in the process of generating the sections.csv file: %s', ex)
-
-
-# Export assigments
-logger.info("Exporting assigments")
-assignments = []
-try:
-    section_ids = map(lambda x: x["id"], sections_list)
-
-    assignments = request_client.get_assignments_by_section_ids(list(section_ids))
-
-    filtered_assignments = [
-        assignment
-        for assignment in assignments
-        if assignment["grading_period"] in grading_periods_array
-    ]
-
-    export_data.to_csv(
-        filtered_assignments, os.path.join(schoology_output_path, "assignments.csv")
-    )
-except Exception as ex:
-    logger.error('An exception has occurred in the process of generating the assigments.csv file: %s', ex)
-
-
-# Export submissions
-logger.info("Exporting submissions")
-try:
-    submissions_list: List[Any] = []
-
-    for assignment in assignments:
-        submissions_response = (
-            request_client.get_submissions_by_section_id_and_grade_item_id(
-                assignment["section_id"],
-                str(assignment["grade_item_id"]),
-                page_size
-            )
+# TODO: this method should disappear when we finish converting all of the output
+# to use the official CSV formats.
+def _create_file_from_list(action: Callable, file_name):
+    logger.info(f"Exporting {file_name}")
+    try:
+        data = action()
+        export_data.to_csv(data, os.path.join(schoology_output_path, file_name))
+    except Exception as ex:
+        logger.error(
+            f"An exception occurred while generating {file_name} : %s",
+            ex,
         )
-        while True:
-            submissions_list = submissions_list + submissions_response.current_page_items
-            if submissions_response.get_next_page() is None:
-                break
-
-    export_data.to_csv(
-        submissions_list, os.path.join(schoology_output_path, "submissions.csv")
-    )
-except Exception as ex:
-    logger.error('An exception has occurred in the process of generating the submissions.csv file: %s', ex)
+
+
+def _get_users():
+    return service.get_users()
```

Comment:
Is this function necessary?

Reply:
No, it is not necessary. I only created it because it felt weird to have all of the other calls wrapped in a function but not this one, like this:

```python
_create_file_from_dataframe(service.get_users(), "users.csv")
_create_file_from_list(_get_sections, "sections.csv")
_create_file_from_list(_get_assignments, "assignments.csv")
_create_file_from_list(_get_submissions, "submissions.csv")
```

That just looks so strange to me that I preferred giving it a separate function. But I'm not attached to it.

Reply:
Okay, it is fine, I like the idea of consistent calls in the main script, improves readability.